### PR TITLE
Implement an interface for beatmap conversion tests

### PR DESCRIPTION
**Pull request by smoogipoo**
Prereqs:
- [x] #2154

Test cases are constructed by osu!stable using https://github.com/peppy/osu-stable/tree/conversion-mappings . These are read in by osu!lazer, which generates data of its own and compares.

I've put in one beatmap for each ruleset for now, but will gather up sane/insane/random/edge cases for each ruleset individually in further PRs, after already-open issues are resolved.

### CODE REVIEW

**Comment by smoogipoo** [comment]
CI isn't going to pass until #2154. Added as a prereq.

File: osu.Game.Rulesets.Catch/Resources/Testing/Beatmaps/basic-expected-conversion.json [hunk 1]
```diff
@@ -0,0 +1 @@
....0,"Position":178.6452},{"StartTime":22250.0,"Position":213.892181},{"StartTime":22312.0,"Position":218.713028},{"StartTime":22375.0,"Position":240.4715},{"StartTime":22437.0,"Position":239.371887},{"StartTime":22500.0,"Position":261.907257},{"StartTime":22562.0,"Position":314.353119},{"StartTime":22625.0,"Position":299.273376},{"StartTime":22687.0,"Position":356.98288},{"StartTime":22750.0,"Position":339.078552},{"StartTime":22812.0,"Position":377.8958},{"StartTime":22875.0,"Position":398.054047},{"StartTime":22937.0,"Position":398.739441},{"StartTime":23000.0,"Position":407.178467},{"StartTime":23062.0,"Position":444.8687},{"StartTime":23125.0,"Position":417.069977},{"StartTime":23187.0,"Position":454.688477},{"StartTime":23250.0,"Position":428.9612},{"StartTime":23312.0,"Position":441.92807},{"StartTime":23375.0,"Position":439.749878},{"StartTime":23433.0,"Position":455.644684},{"StartTime":23491.0,"Position":440.7359},{"StartTime":23549.0,"Position":430.0944},{"StartTime":23607.0,"Position":420.796173},{"StartTime":23665.0,"Position":435.897461},{"StartTime":23723.0,"Position":418.462555},{"StartTime":23781.0,"Position":405.53775},{"StartTime":23874.0,"Position":408.720825}]}]}
```
Comment (peppy) [inline osu.Game.Rulesets.Catch/Resources/Testing/Beatmaps/basic-expected-conversion.json]: let's prettify this so it will diff well if we make changes

Comment (smoogipoo) [reply osu.Game.Rulesets.Catch/Resources/Testing/Beatmaps/basic-expected-conversion.json]: Not that we should ever be making manual changes to this file, and it's autogenerated.

Comment (peppy) [reply osu.Game.Rulesets.Catch/Resources/Testing/Beatmaps/basic-expected-conversion.json]: it's exactly for that scenario (so when automatic change are made we can see the diff easily).

File: osu.Game/Beatmaps/BeatmapConverter.cs [hunk 2]
```diff
@@ -8,12 +8,36 @@
 
 namespace osu.Game.Beatmaps
 {
+    public interface ITestableBeatmapConverter
```
Comment (peppy) [inline osu.Game/Beatmaps/BeatmapConverter.cs]: Can probably just call this `IBeatmapConverter`, doesn't have anything specific to testing here.

Also move to its own file please.

Comment (smoogipoo) [reply osu.Game/Beatmaps/BeatmapConverter.cs]: I really don't want to expose this as an `IBeatmapConverter`, if not for the fact that it doesn't give you a resulting `Beatmap`, just an event to consume.

**Review by peppy — DISMISSED** [review]

**Review by peppy — APPROVED** [review]

**Pull request merged**